OnBaseDocsApi/Controllers/OnBaseDocsController.cs

```diff
@@ -55,7 +59,189 @@ public IHttpActionResult GetFile(int id)
+                return BadRequestResult("Uploading a document requires a multi-part POST request.");
+            }
+
+            Stream docStream = null;
+            string docExt = null;
+            DocumentPostAttributes docAttr = null;
+
+            try
+            {
+                var provider = await Request.Content.ReadAsMultipartAsync(
+                    new MultipartMemoryStreamProvider());
+
+                foreach (HttpContent content in provider.Contents)
+                {
+                    // The Content-Disposition header is required.
+                    if (content.Headers.ContentDisposition == null)
+                        return BadRequestResult($"A Content-Disposition header is required.");
+
+                    var dispoName = content.Headers.ContentDisposition.Name.Trim('"');
+
+                    if (dispoName == "file")
+                    {
+                        // The content is a file.
+                        // Only allow one content file.
+                        if ((docStream != null) || !string.IsNullOrEmpty(docExt))
+                            return BadRequestResult("More than one document content was included.");
+
+                        var fileName = content.Headers.ContentDisposition.FileName.Trim('"');
+                        docStream = content.ReadAsStreamAsync().Result;
+                        docExt = Path.GetExtension(fileName).TrimStart('.');
+                    }
+                    else if (dispoName == "attributes")
+                    {
+                        // The content is JSON.
+                        // Only allow one JSON attributes.
+                        if (docAttr != null)
+                            return BadRequestResult("More than one document attributes was included.");
+
+                        docAttr = JObject
+                            .Parse(content.ReadAsStringAsync().Result)
+                            .ToObject<DocumentPostAttributes>();
+                    }
+                    else
+                    {
+                        return BadRequestResult($"Unexpected Content-Disposition header of name '{dispoName}'.");
+                    }
+                }
+            }
+            catch (JsonReaderException ex)
+            {
+                return BadRequestResult($"JSON parse error. {ex.Message}");
+            }
+            catch (IOException ex) when (ex.Message.Contains("MIME multipart message is not complete"))
+            {
+                return BadRequestResult(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                return InternalErrorResult(ex.Message);
+            }
+
+            if ((docStream == null) || string.IsNullOrWhiteSpace(docExt))
+                return BadRequestResult("The required parameter document content is missing.");
+            if (docAttr == null)
+                return BadRequestResult("The required parameter document attributes are missing.");
```

Comment:
For these messages, could you reference the required keys in the request (`file` and `attributes`)?

Reply:
Good catch. Done.